api/service/projects_service.go

```diff
@@ -116,17 +124,43 @@ func (service *projectsService) ListProjects(ctx context.Context, name string, u
 	return allProjects, nil
 }
 
-func (service *projectsService) UpdateProject(ctx context.Context, project *models.Project) (*models.Project, error) {
+func (service *projectsService) UpdateProject(ctx context.Context, project *models.Project) (*models.Project,
+	map[string]interface{}, error) {
+	existingProject, err := service.FindByID(project.ID)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error fetching project with id %s: %w", project.ID, err)
+	}
+
 	if service.authEnabled {
 		err := service.updateAuthorizationPolicy(ctx, project)
 		if err != nil {
-			return nil, fmt.Errorf("error while updating authorization policy for project %s", project.Name)
+			return nil, nil, fmt.Errorf("error while updating authorization policy for project %s", project.Name)
 		}
 	}
+
+	if isLabelBlacklisted(existingProject.Labels, project.Labels, service.updateProjectConfig.LabelsBlacklist) {
```

Comment:
https://github.com/caraml-dev/mlp/pull/99#discussion_r1636089464 Not sure if you skipped this comment because it's now outdated but I don't think they have been resolved yet.